src/Agda2Hs/Compile/Term.hs

```diff
@@ -286,7 +286,9 @@ compileTerm v = do
         body <- compileTerm body
         return $ case body of
           Hs.InfixApp _ a op b
-            | a == hsx -> Hs.RightSection () op b -- System-inserted visible lambdas can only come from sections
+            | a == hsx
+            , pp op /= "-"             -- Jesper: no right section for minus, as Haskell parses this as negation!
+            -> Hs.RightSection () op b -- System-inserted visible lambdas can only come from sections
```

Comment:
Introducing `\ section -> section - ...` is quite ugly; why not use a left section on [subtract = flip (-)](https://hackage.haskell.org/package/base-4.7.0.2/docs/Prelude.html#v:subtract) instead?
```hs
            | a == hsVar x
            -> if pp op == "-" then
                   -- no right section for minus, as Haskell parses this as negation!
                   Hs.LeftSection () (QConOp $ unQual "subtract") b 
                else
                   -- System-inserted visible lambdas can only come from sections
                   Hs.RightSection () op b
```

Minor comments:
1. No need for the intermediate variable `hsx`, just inline here
2. No need to mention the name of the person writing the comment :P

Reply:
For this PR I wasn't really looking at whether the output is *pretty*, just whether it is *correct*. But I agree that your version is prettier.

The variable `hsx` was already there, I didn't want to touch it since it's unrelated to this PR.

Writing my name in comments is a habit I got from working on Agda itself where it seems to be standard practice. I personally quite like these little bits of personality in the codebase :)